src/lib.rs

```diff
@@ -242,6 +242,42 @@ pub struct Histogram<T: Counter> {
     counts: Vec<T>,
 }
 
+/// Errors that can occur when creating a histogram.
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+pub enum CreationError {
+    /// Lowest discernible value must be >= 1
+    LowTooSmall,
+    /// Lowest discernible value must be <= u64::max_value() / 2
+    LowTooBig,
+    /// Highest trackable value must be >= 2 * lowest discernible value
+    HighLessThanTwiceLow,
+    /// Number of significant digits must be between 0 and 5
+    SigFigTooBig,
+    /// Cannot represent sigfig worth of values beyond low
+    CannotRepresentSigFigBeyondLow
+}
+
+/// Errors that can occur when adding another histogram.
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+pub enum AdditionError {
+    /// The other histogram includes values that do not fit in this histogram's range.
+    /// Only possible when auto resize is disabled.
+    OtherAddendValuesExceedRange
```

Comment:
Annoyingly, "addend" refers to both terms in an addition. The first one is called the "augend" but that's not what we care about. Oddly this value thing means that addition of non-resizing histograms is not commutative.